packages/driver/src/cy/commands/actions/check.ts

```diff
@@ -113,6 +113,13 @@ const checkOrUncheck = (Cypress, cy, type, subject, values: any[] = [], userOpti
           cy.ensureVisibility($el, options._log)
         }
 
+        // if the checkbox is in an indeterminate state, checking or unchecking should set the
+        // prop to false to move it into a "determinate" state
+        // https://github.com/cypress-io/cypress/issues/19098
+        if ($el.prop('indeterminate')) {
+          $el.prop('indeterminate', false)
+        }
+
```

Comment:
So interesting note: `cy.uncheck()` on an indeterminate checkbox doesn't emit a click event.

At first I thought this was a bug, but the more I think about it, the more it seems like the correct behavior. Users can't uncheck indeterminate checkboxes - clicking on them checks them, which was a case that already worked as expected.

**Indeterminate checkboxes are already checked or unchecked**, and removing indeterminate on what would otherwise be a no-op is purely a developer QOL feature. There's no browser behavior to emulate, so not emitting any events seems correct.